app/controllers/order_controller.rb

```diff
@@ -0,0 +1,133 @@
+class OrderController < ApplicationController
+  before_action :authenticate_user!
+  before_action :current_user_cart
+
+  def index
+    @orders = current_user_orders.order(created_at: :desc)
+  end
+
+  def show
+    @order = current_user_orders.find_by(id: params[:id])
+  end
+
+  def checkout
+    @books = @cart.books
+    @total_price = @books.sum(&:price)
+  end
+
+  def payment
+    begin
+      ActiveRecord::Base.transaction do
+        books = @cart.books
+        total_price = books.sum(&:price)
+        CartItem.where(cart_id: @cart.id).delete_all
+
+        @order = current_user.orders.new(
+          total_price: total_price,
+          status: :pending,
+          shipping_address: order_params[:shipping_address]
+        )
+
+        books.each do |book|
+          @order.order_books.new(
+            book: book,
+            book_title: book.title,
+            price_per_unit: book.price,
+            quantity: 1
+          )
+        end
+
+        @order.save!
+
+        payment_intent = Stripe::PaymentIntent.create(
+          amount: (total_price * 100).to_i,
+          currency: Constants::CURRENCY,
+          payment_method_types: Constants::STRIPE_PAYMENT_METHODS,
+          metadata: {
+            order_id: @order.id,
+            user_id: current_user.id,
+            book_ids: books.ids.join(",")
+          }
+        )
+        @client_secret = payment_intent.client_secret
+        @stripe_publishable_key = ENV["STRIPE_PUBLISHABLE_KEY"]
+        @order.stripe_payment_intent = payment_intent.id
+        @order.save!
+      end
+    rescue => e
```

Comment:
Do we need to specify which error we want to rescue?

Reply:
Yes, we do. I have some ideas to rescue. 
``` ruby
    rescue Stripe::InvalidRequestError => e
      err_code = e.json_body[:error][:code]
      case err_code
      when "amount_too_small"
        err_message = "Min. amount must be $0.50"
      else
        err_message = e.message
      end
      redirect_to order_checkout_path, alert: err_message
    rescue => e
      redirect_to order_checkout_path, alert: e.message
    end
```

Do you have any suggestions about the errors we should cover here?

Reply:
I think we should only rescue the error we expect. For error we don't expect, it's better to raise exception so that we can get notified using services such as [HoneyBadger](https://www.honeybadger.io/), AppSignal(https://www.appsignal.com/) or [Rollbar](https://rollbar.com/)

Reply:
Thank you for your suggestions. I tried [HoneyBadger](https://www.honeybadger.io/), it's so helpful.